server/v2/controllers/userController.js

```diff
@@ -0,0 +1,65 @@
+import pool from '../config/dbConnect';
+import User from '../models/userModel';
+import queries from '../models/queries';
+import Auth from '../helpers/authenticate';
+
+class UserController {
+    static async signUp(req, res) {
+        try {
+            const existingEmail = await pool.query(queries.findOneUser, [req.body.email]);
+            if (existingEmail.rowCount === 1) {
+                return res.status(409).json({
+                    status: 409,
+                    error: 'Email already exists!'
+                });
+            }
+
+            const existingUsername = await pool.query(queries.findOneUsername, [req.body.username]);
```

Comment:
Move database related operations to the relevant Model; for instance, the methods to create, update, delete and get user info should all be located in user Model.